lib/livebook/hubs/personal.ex

```diff
@@ -131,4 +147,54 @@ defimpl Livebook.Hubs.Provider, for: Livebook.Hubs.Personal do
   end
 
   def connection_error(_personal), do: raise("not implemented")
+
+  def notebook_stamp(_hub, _notebook_source, metadata) when metadata == %{} do
+    :skip
+  end
+
+  def notebook_stamp(hub, notebook_source, metadata) do
+    # We use AES-GCM-128 to encrypt metadata and generate signature
+    # for both metadata and the notebook source. We make use of the
+    # implementation in MessageEncryptor, which conveniently returns
+    # a single token
+
+    {secret, sign_secret} = derive_keys(hub.secret_key)
+
+    payload = :erlang.term_to_binary(metadata)
+    notebook_source = IO.iodata_to_binary(notebook_source)
```

Comment:
Can you please send a PR to the encryptor so it allows AAD to be a binary or a list? We should be able to allow iolists there, because Erlang allows iolists in crypto.

Reply:
Sounds good!

Reply:
Nevermind, I will do it now quickly. :)

Reply:
Plug Crypto 1.2.5 is out. Please give it a try:

```suggestion
```

Reply:
Works perfectly!!